trieste/models/keras/data.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+
+from trieste.data import Dataset
+from trieste.models.keras import KerasNetwork
+from trieste.models.optimizer import TrainingData
+
+
+class EnsembleDataTransformer:
+    """
+    A `DatasetTransformer` for Keras ensemble networks.
+    """
+    def __init__(self, networks: List[KerasNetwork]):
+        self._networks = networks
+
+    def __call__(self, data: Dataset) -> TrainingData:
+        inputs = {}
+        outputs = {}
+
+        for index, network in enumerate(self._networks):
+            input_name = network.input_tensor_spec.name + "_" + str(index)
+            if index == 0:
+                output_name = 'dense'
```

Comment:
This isn't going to work in general. We will need to figure out how to get the input and output names from the model.

Reply:
not sure I understand completely why naming is so relevant, but its definitely a good idea to have a dedicated class for these trieste Dataset to keras data tranformations